lib/rspec/mocks/object_reference.rb

```diff
@@ -42,44 +42,67 @@ def self.anonymous_module?(mod)
     # Used when an object is passed to `object_double`.
     # Represents a reference to that object.
     #
-    # @private
+    # @param object
```

Comment:
One other thing that I think will really help clarify the purpose and use of these classes is a mention in the class docs of each implementation that each of these is an implementation of a common abstract interface, and then you can use YARD's `# @see AlternateClass` syntax to reference the other implementations.